main.py

```diff
@@ -1,53 +1,223 @@
+
+    Args:
+        model (torch.nn.Module): The trained model to be evaluated.
+        test_dataloader (DataLoader): The test dataset loader containing notes and labels in batches.
+
+    Returns:
+        Returns:
+        tuple: A tuple containing the evaluation results.
+            - val_loss (float): The average loss of the model on the test dataset.
+            - val_accuracy (float): The accuracy of the model on the test dataset, as a percentage.
+    """
+    # get test data
+    criterion = nn.CrossEntropyLoss
+    loss_sum = 0
+    correct = 0
+    total = 0
+    with torch.no_grad():
+        for data in test_dataloader:
+            notes, labels = data
+            total += labels.size(0)
+            out = model(notes)
+            loss = criterion(out, labels)
+            loss_sum += loss.item()
+            _, preds = torch.max(out, 1)
+            for pred, label in zip(preds, labels):
+                if pred == label:
+                    correct += 1
+    val_loss = loss_sum / len(test_dataloader)
+    val_accuracy = correct / total
+    return val_loss, val_accuracy
+
+
+def test_model(model: nn.Module, test_data: DataLoader, path: Optional[str] = None, classnames=None):
+    """
+    Evaluate the performance of the trained ComposerClassifier model with the provided test data loader.
+
+    Args:
+        model (PitchSeqNN): The trained ComposerClassifier model to be evaluated.
+        test_data (DataLoader): Data for the model to be tested on
+        path (str, optional): If specified, the function will load the model's state from the
+                              provided path.
+        classnames (list[str], optional): Classnames for printing results.
+    """
+    if classnames is None:
+        classnames = [0, 1]
+    # if path is specified load state from file
+    if path is not None:
+        model.load_state_dict(torch.load(path))
+    # containers for counting prediction
+    correct_pred = {classname: 0 for classname in classnames}
+    total_pred = {classname: 0 for classname in classnames}
+    correct = 0
+    total = 0
+    with torch.no_grad():
+        for data in test_data:
+            notes, labels = data
+            out = model(notes)
+            loss = nn.CrossEntropyLoss()(out, labels)
+            print(loss)
+            _, preds = torch.max(out, 1)
+            correct += (preds == labels).sum().item()
+            total += labels.size(0)
+
+            for label, pred in zip(labels, preds):
+                if label == pred:
+                    correct_pred[classnames[label]] += 1
+                total_pred[classnames[label]] += 1
+    for classname, correct_count in correct_pred.items():
+        accuracy = 100 * float(correct_count) / total_pred[classname]
+        print(f"Accuracy for class: {classname:5s} is {accuracy:.1f} %")
+    print(f"Accuracy of the network on the test data: {(100 * correct / total):.3f} %")
+    print(f"correctly predicted : {correct_pred}")
```

Comment:
I think using a [confusion matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html) here would work better. Your version could get problematic to maintain if we were to start adding classes.